activerecord/test/cases/adapters/postgresql/enum_test.rb

```diff
@@ -197,4 +186,55 @@ def test_enum_type_explicit_schema
   ensure
     @connection.drop_schema("test_schema", if_exists: true)
   end
+
+  def test_schema_dump_scoped_to_schemas
+    with_test_schema("test_schema") do
+      @connection.create_enum("mood_in_test_schema", ["sad", "ok", "happy"])
+      @connection.create_table("postgresql_enums_in_test_schema") do |t|
+        t.column :current_mood, :mood_in_test_schema
+      end
+
+      output = dump_all_table_schema
+
+      assert output.include?('create_enum "public.mood", ["sad", "ok", "happy"]'), output
+      assert output.include?('create_enum "mood_in_test_schema", ["sad", "ok", "happy"]'), output
+      assert output.include?('t.enum "current_mood", enum_type: "mood_in_test_schema"'), output
+    end
+  end
+
+  def test_schema_load_scoped_to_schemas
+    original, $stdout = $stdout, StringIO.new
+
+    with_test_schema("test_schema", drop: false) do
+      ActiveRecord::Schema.define do
+        create_enum "mood_in_test_schema", ["sad", "ok", "happy"]
+        create_enum "public.mood", ["sad", "ok", "happy"]
+
+        create_table "postgresql_enums_in_test_schema", force: :cascade do |t|
+          t.enum "current_mood", enum_type: "mood_in_test_schema"
+        end
+      end
+
+      assert @connection.column_exists?(:postgresql_enums_in_test_schema, :current_mood, sql_type: "mood_in_test_schema")
+    end
+
+    # This is outside `with_test_schema`, so we need to explicitly specify which schema we query
+    assert @connection.column_exists?("test_schema.postgresql_enums_in_test_schema", :current_mood, sql_type: "test_schema.mood_in_test_schema")
+  ensure
+    @connection.drop_schema("test_schema")
+    $stdout = original
+  end
+
+  private
+    def with_test_schema(name, drop: true)
+      old_search_path = @connection.schema_search_path
+      @connection.create_schema(name)
+      @connection.schema_search_path = name
+      @connection.schema_cache.clear!
```

Comment:
Not sure, but I think we do not need to clear cache here and in the ensure block.